modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -87,15 +97,24 @@ def report msg, response, error_only
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
 
-    def match_zone(fqdn)
+    def match_zone(fqdn_or_ip)
```

Comment:
You increased ```match_zone```'s cyclomatic complexity, made it responsible for more than one thing, relied on tricky call semantics, made testing harder -- not a good change.